pkg/kubernetes/scripts/policy.js

```diff
@@ -174,6 +177,66 @@
                 return result;
             }
 
+            var sarCache = { };
+
+            function subjectAccessReview(project, user, verb, resource) {
+                var key = (project || "") + ':' + (user ? user.metadata.name : "")+ ':' + verb + ':' + resource;
+                // cached already?
+                if (key in sarCache)
+                    return sarCache[key];
+
+                // pre-fill cache to avoid multiple concurrent requests
+                sarCache[key] = false;
+
+                loader.watch("Projects").then(function() {
+                    // if no project was given, check the first one as an approximation
+                    // TODO: is that sufficient, or should we iterate over all
+                    // projects? (much more expensive)
+                    if (user && !project) {
+                        var projects = select().kind("Project");
+                        for (var p in projects) {
+                            project = projects[p].metadata.name;
+                            break;
+                        }
+                        if (!project)
+                            return false; // the user has no projects
+                    }
+
+                    var request = {
+                        kind: "SubjectAccessReview",
+                        apiVersion: "v1",
+                        namespace: project,
+                        verb: verb,
+                        resource: resource
+                    };
+
+                    methods.post(loader.resolve("subjectaccessreviews"), request)
+                        .then(function(response) {
+                            sarCache[key] = response.allowed;
+                            // tell UI to update only if allowed, as our "no lookup yet" default is false already
+                            if (response.allowed) {
+                                $rootScope.$applyAsync();
```

Comment:
I'd rather not have to apply scope in here. Is there a way we can this stick to generating the result from the promise and have something else in the dashboard code that stores the value on a property that is used in the template.